### PR TITLE
add element type parameter to VTree and introduce alias VNodeT

### PR DESCRIPTION
**Pull request by cornerman**

### CODE REVIEW

File: src/main/scala/outwatch/dom/package.scala [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 import cats.effect.IO
+import org.scalajs.dom.Element
 
 package object dom extends Implicits with ManagedSubscriptions with SideEffects with MonixOps {
 
   val Sink = outwatch.Sink
   val Handler = outwatch.Handler
 
-  type VNode = IO[VTree]
+  type VNodeT[+T <: Element] = IO[VTree[T]]
```
Comment (LukaJCB) [inline src/main/scala/outwatch/dom/package.scala]: I don't think I like the `VNodeT` name. It uses the same convention as Monad transformers in cats, which might confuse done people familiar with them 😁

Comment (cornerman) [reply src/main/scala/outwatch/dom/package.scala]: True, that name is not really intuitive. Do you have a better name? :)

**Comment by cornerman** [comment]
There is one minor annoyance, you cannot write this anymore:
```scala
val handler: Handler[VNode] = Handler.create(div("hallo"))
```

Because Handler is invariant in its type param, we need write one of these two versions:
```scala
val handler: Handler[VNode] = Handler.create[VNode](div("hallo"))
// or
val handler: Handler[VNodeT[html.Div]] = Handler.create(div("hallo"))
```

File: src/main/scala/outwatch/dom/package.scala (hunk 1 above)
Comment (cornerman) [reply src/main/scala/outwatch/dom/package.scala]: @LukaJCB @fdietze What would be a good name for `VNodeT`?

Comment (fdietze) [reply src/main/scala/outwatch/dom/package.scala]: `TypedVNode`?

Comment (fdietze) [reply src/main/scala/outwatch/dom/package.scala]: `GenericVNode`?

**Comment by fdietze** [comment]
> There is one minor annoyance, you cannot write this anymore:
> 
> ```scala
> val handler: Handler[VNode] = Handler.create(div("hallo"))
> ```

I think this use-case is very rare. It should be fine.

**Pull request closed without merging**